docs/source/api/python/mantid/geometry/DetectorInfo.rst

```diff
@@ -0,0 +1,154 @@
+==============
+ DetectorInfo
+==============
+
+This is a python binding to the C++ class Mantid::Geometry::DetectorInfo.
+
+--------
+Purpose
+--------
+The purpose of the ``DetectorInfo`` object is to allow the user to access information about the detector(s) being used in an experiment. The ``DetectorInfo`` object can be used to access geometric information such as the number of detectors in the beamline, the absolute position of a detector as well as the absolute rotation of a detector.
+
+Many users may need this extra information so that they can have a better understanding of the beamline they are using. This information is easy and fast to access. Some information like mask flags can be modified directly.
+
+The ``DetectorInfo`` object is one of three objects that the user can gain access to from a workspace. 
+The other two are:
+ * ``SpectrumInfo``
+ * ``ComponentInfo``
+
+---------
+Indexing
+---------
+The ``DetectorInfo`` object is accessed by an index going from 0 to N-1, where N is the number of detectors.
+A detector index is a way of addressing and enumerating detectors in the beamline.
+A detector index can be found from a detector ID using ``indexOf``. 
+
+-------
+Usage
+-------
+
+**Example 1 - Creating a DetectorInfo Object:**
+This example shows how to obtain a ``DetectorInfo`` object from a workspace object.
+The return value is a ``DetectorInfo`` object.
+
+.. testcode:: CreateDetectorInfoObject
+	
+	# Create a workspace to use
+	ws = CreateSampleWorkspace()
+
+	# Get the DetectorInfo object
+	info = ws.detectorInfo()
+	print(type(info))
+
+Output:
+
+.. testoutput:: CreateDetectorInfoObject
+
+	<class 'mantid.geometry._geometry.DetectorInfo'>
+
+
+**Example 2 - Calling the setMasked method on the DetectorInfo Object:**
```

Comment:
Is having separate examples for each method useful? Maybe combining them gives a more concise overview?

Reply:
I thought having smaller self contained examples would be better for users? Also. each of the tests has a short description of what the method being called does - if everything is combined into one example then should we just remove the descriptions?

Reply:
Yes, small is good, but my feeling was the the ratio between "code for setting up and commenting" and "actually interesting code" was not so favorable. One or two very basic examples are definitely good (you can for example combine example 1 and 2 into a basic example), but then I would have one or two more realistic ones (maybe one for read access, another one for write access), with a loop over all detectors or spectra, accessing a couple of methods in the loop body.